R/compat.r

```diff
@@ -18,3 +18,25 @@ warn_unicode_on_windows <- function(code, warn) {
         if (real_len != r_len) warn(UNICODE_WARNING)
     }
 }
+
+
+DEVICE_ERR <-
+"Your R installation is misconfigured:
+getOption('bitmapType') == '%s', but capabilities('%s') == FALSE.
+This means that the png() device is unable to work.
+Please %s."
+
+preflight_check <- function() {
+    bitmap_type <- getOption('bitmapType')
+    xlib  <- capabilities('X11')
+    cairo <- capabilities('cairo')
+    
+    insane_xlib  <- bitmap_type == 'Xlib'  && !xlib
+    insane_cairo <- bitmap_type == 'cairo' && !cairo
+    if (insane_xlib || insane_cairo) {
+        help_msg <-
+            if (!xlib && !cairo) 'install an R that is configured to have either Xlib or cairo support'
+            else sprintf("put `options(bitmapType='%s')` into your ~/.Rprofile", if (xlib) 'Xlib' else 'cairo')
```

Comment:
If we know one of them works, can we just switch to that instead of complaining to the user? If it's running on a remote server (e.g. in Jupyterhub), they may well never see this error message, just the kernel dying.